tests/test_clock.py

```diff
@@ -69,6 +69,50 @@ def test_timer_duration_not_matched_or_exceeded(self):
         ]  # yapf: disable
         self.assertFalse(clock.Timer(self.mock_clock, duration).expired())
 
+    def test_set_remaining_fails_when_remaining_is_longer_than_duration(self):
+        duration = datetime.timedelta(days=3)
+        self.mock_clock.now.return_value = datetime.datetime(
+            2000, 1, 1, 0, 0, 0, 0, tzinfo=pytz.utc)
+        timer = clock.Timer(self.mock_clock, duration)
+        with self.assertRaises(ValueError):
+            timer.set_remaining(datetime.timedelta(days=4))
+
+    def test_set_remaining_fails_when_remaining_is_negative(self):
+        duration = datetime.timedelta(days=3)
+        self.mock_clock.now.return_value = datetime.datetime(
+            2000, 1, 1, 0, 0, 0, 0, tzinfo=pytz.utc)
+        timer = clock.Timer(self.mock_clock, duration)
+        with self.assertRaises(ValueError):
+            timer.set_remaining(datetime.timedelta(days=-1))
+
+    def test_set_remaining_succeeds_when_remaining_is_zero(self):
+        duration = datetime.timedelta(days=3)
+        self.mock_clock.now.return_value = datetime.datetime(
+            2000, 1, 1, 0, 0, 0, 0, tzinfo=pytz.utc)
+        timer = clock.Timer(self.mock_clock, duration)
+        timer.set_remaining(datetime.timedelta(seconds=0))
+        self.assertTrue(timer.expired())
+
+    def test_set_remaining_succeeds_when_remaining_equals_duration(self):
```

Comment:
Oh, I actually just meant that ```timer._end_time``` doesn't change when the duration is equal to ```time_remaining```. But, now that I type that out I realize that's because we're actually using the initial ```clock.now()``` return value twice, once when ```reset()``` is called upon initialization, and once when ```set_remaining()``` is called.